streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -127,9 +127,15 @@ public String queryableStoreName() {
             }
             // we can inherit parent key and value serde if user do not provide specific overrides, more specifically:
             // we preserve the key following the order of 1) materialized, 2) parent
-            keySerde = materializedInternal.keySerde() != null ? materializedInternal.keySerde() : this.keySerde;
+            if (materializedInternal.keySerde() == null) {
+                materializedInternal.withKeySerde(this.keySerde);
+            }
+            keySerde = materializedInternal.keySerde();
             // we preserve the value following the order of 1) materialized, 2) parent
-            valueSerde = materializedInternal.valueSerde() != null ? materializedInternal.valueSerde() : this.valSerde;
+            if (materializedInternal.valueSerde() == null) {
```

Comment:
This is for 2) as well: Materialized did not use the inherited if it was not specified.